rbac/management/group/view.py

```diff
@@ -579,3 +568,43 @@ def validate_and_get_exclude_key(self, params):
                 exclude)
             raise serializers.ValidationError({key: _(message)})
         return exclude
+
+    def obtain_principals(self, request, group):
+        """Obtain principals based on request, supports exclusion."""
+        principals_from_params = self.filtered_principals(group, request)
+
+        exclude = self.validate_and_get_exclude_key(request.query_params)
+
+        if exclude == 'false':
+            page = self.paginate_queryset(principals_from_params)
+            serializer = PrincipalSerializer(page, many=True)
+        else:
+            # Get all the principals to be excluded
+            serializer = PrincipalSerializer(principals_from_params, many=True)
+
+        principal_data = serializer.data
+        if principal_data:
+            username_list = [principal['username'] for principal in principal_data]
+        else:
+            username_list = []
+
+        # Call BOP to get principals
+        proxy = PrincipalProxy()
+
+        if exclude == 'false':
+            resp = proxy.request_filtered_principals(username_list, request.user.account)
+        else:
+            resp = proxy.request_principals(request.user.account, TOTAL_PRINCIPALS)
+
+        # Deal with error
+        if isinstance(resp, dict) and 'errors' in resp:
+            return Response(status=resp.get('status_code'), data=resp.get('errors'))
+
+        if exclude == 'false':
+            response_data = resp.get('data')
+        else:
+            data_without_pagination = [principal for principal in resp.get('data').get('users')
```

Comment:
Ah, maybe I spoke too soon. 😄I assume this is what handles the pagination on our end?

Reply:
Ah, right, this is the one that handles pagination of the data after exclusion.